google/cloud/storagecontrol/v2/samples/storage_control_folder_samples.cc

```diff
@@ -0,0 +1,219 @@
+                  std::vector<std::string> const& argv) {
+  // [START storage_control_create_folder]
+  namespace storagecontrol = google::cloud::storagecontrol_v2;
+  [](storagecontrol::StorageControlClient client,
+     std::string const& bucket_name, std::string const& folder_id) {
+    auto const parent = std::string{"projects/_/buckets/"} + bucket_name;
+    auto folder = client.CreateFolder(
+        parent, google::storage::control::v2::Folder{}, folder_id);
+    if (!folder) throw std::move(folder).status();
+
+    std::cout << "Created folder: " << folder->name() << "\n";
+  }
+  // [END storage_control_create_folder]
+  (std::move(client), argv.at(0), argv.at(1));
+}
+
+void DeleteFolder(google::cloud::storagecontrol_v2::StorageControlClient client,
+                  std::vector<std::string> const& argv) {
+  // [START storage_control_delete_folder]
+  namespace storagecontrol = google::cloud::storagecontrol_v2;
+  [](storagecontrol::StorageControlClient client,
+     std::string const& bucket_name, std::string const& folder_id) {
+    auto const name = std::string{"projects/_/buckets/"} + bucket_name +
+                      std::string{"/folders/"} + folder_id;
+    auto status = client.DeleteFolder(name);
+    if (!status.ok()) throw std::move(status);
+
+    std::cout << "Deleted folder: " << folder_id << "\n";
+  }
+  // [END storage_control_delete_folder]
+  (std::move(client), argv.at(0), argv.at(1));
+}
+
+void GetFolder(google::cloud::storagecontrol_v2::StorageControlClient client,
+               std::vector<std::string> const& argv) {
+  // [START storage_control_get_folder]
+  namespace storagecontrol = google::cloud::storagecontrol_v2;
+  [](storagecontrol::StorageControlClient client,
+     std::string const& bucket_name, std::string const& folder_id) {
+    auto const name = std::string{"projects/_/buckets/"} + bucket_name +
+                      std::string{"/folders/"} + folder_id;
+    auto folder = client.GetFolder(name);
+    if (!folder) throw std::move(folder).status();
+
+    std::cout << "Got folder: " << folder->name() << "\n";
+  }
+  // [END storage_control_get_folder]
+  (std::move(client), argv.at(0), argv.at(1));
+}
+
+void ListFolders(google::cloud::storagecontrol_v2::StorageControlClient client,
+                 std::vector<std::string> const& argv) {
+  // [START storage_control_list_folders]
+  namespace storagecontrol = google::cloud::storagecontrol_v2;
+  [](storagecontrol::StorageControlClient client,
+     std::string const& bucket_name) {
+    auto const parent = std::string{"projects/_/buckets/"} + bucket_name;
+    for (auto folder : client.ListFolders(parent)) {
+      if (!folder) throw std::move(folder).status();
+      std::cout << folder->name() << "\n";
+    }
+
+    std::cout << bucket_name << std::endl;
+  }
+  // [END storage_control_list_folders]
+  (std::move(client), argv.at(0));
+}
+
+google::cloud::Status RemoveStaleFolders(
```

Comment:
Out of curiosity, why do we need this?

Reply:
Because the tests can crash before cleaning up the resources (folders in this case) they create.